go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -875,17 +876,13 @@ func (p *DoltDatabaseProvider) registerNewDatabase(ctx *sql.Context, name string
 		}
 	}
 
-	mrEnv, err := env.MultiEnvForSingleEnv(ctx, newEnv)
-	if err != nil {
-		return err
-	}
-	dbs, err := ApplyReplicationConfig(ctx, sql.NewBackgroundThreads(), mrEnv, cli.CliErr, db)
+	sdb, err := applyReadReplicationConfigToDatabase(ctx, newEnv, db)
```

Comment:
`ApplyReplicationConfig` creates the commit hooks and is responsible for the transform of the dsess.SqlDatabase into a ReadOnlyDatabase if it needs to be configured as such. Here, we just need the later functionality: the other functionality is already handled (and more) by our InitDatabaseHook.